src/sparseml/modifiers/quantization/gptq/base.py

```diff
@@ -0,0 +1,160 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import Any, Dict, List, Optional, Union
+
+from sparseml.core import Modifier
+from sparseml.core.factory import ModifierFactory
+from sparseml.core.model.base import ModifiableModel
+from sparseml.core.state import State
+
+
+__all__ = ["GPTQModifier"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class GPTQModifier(Modifier):
+    """
+    Modifier for applying the one-shot OBCQ algorithm to a model
+
+    Lifecycle:
+        - on_initialize
+            - initialize_compression()
+                - compressible_layers()
+                - LayerCompressor.pre_compress()
+            - apply_compression()
+                - run_calibration_forward()
+                - LayerCompressor.compress()
+                - LayerCompressor.post_compress()
+                - LayerCompressor.revert_layer_wrappers()
+        - on_finalize
+
+    :param sequential_update: Whether or not to update weights sequentially by layer,
+        True saves on GPU memory
+    :param targets: list of layer names to compress during GPTQ, or '__ALL__'
+        to compress every layer in the model
+    :param block_size: Used to determine number of columns to compress in one pass
+    :param quantize: Whether or not to quantize weights. Set to
+        True to quantize using an existing quantization modifier, or pass in the
+        configuration for a quantization modifier if one does not already exist
+        in the recipe
+    :param dampening_frac: Amount of dampening to apply to H, as a fraction of the
+        diagonal norm
+    """
+
+    sequential_update: Optional[bool] = False
+    targets: Union[str, List[str], None] = None
+    compressible_layers_: Optional[List] = None
+    block_size: int = 128
+    quantize: Union[bool, Dict] = True
```

Comment:
Can't think of a case where we would want this to be false, can we just remove this arg? We should always be quantizing when calling this module

Reply:
You're correct, we will always be quantizing when calling this module, I've kept it here more for accepting a quant modifier like we did before, it does make sense to remove this in refactor, I will open a subsequent PR to do that